### PR TITLE
[ADVAPP-323]: Introduce the ability to set an icon for knowledge base categories

### PR DESCRIPTION
**Pull request by danharrin**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-323

### Technical Description

Uses the Blade icons factory and caches the results. Using a separate Blade view for options to allow Tailwind to discover the classes, and I feel too icky putting CSS classes in PHP classes.

### Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Content or styling update (Changes which don't affect functionality)
- [ ] DevOps
- [ ] Documentation

### Screenshots (if appropriate)

https://github.com/canyongbs/advisingapp/assets/41773797/852b8de2-eb95-40b2-8d54-e9cd202e3bab

### Any deployment steps required?

A deployment step could be a command that needs to be executed or an ENV key that needs to be added, for example.

- [ ] Yes, please specify
- [x] No

_______________________________________________

## Before contributing and submitting this PR, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/canyongbs/advisingapp/blob/main/README.md#contributing).
* [x] Title the PR with the ticket/issue number and a short description of the changes made. Or if no ticket/issue exists, title the PR with a short description of the changes made
* [x] Linked a relevant ticket or issue or describe the issue/feature which this PR resolves/implements.
* [x] Resolved all conflicts, if any.
* [x] Rebased your branch PR on top of the latest upstream `main` branch.

### CODE REVIEW

File: app-modules/knowledge-base/database/migrations/2024_02_12_123144_add_icon_column_to_knowledge_base_categories_table.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        Schema::table('knowledge_base_categories', function (Blueprint $table) {
+            $table->string('icon')->nullable()->after('description');
```
Comment (Orrison) [inline app-modules/knowledge-base/database/migrations/2024_02_12_123144_add_icon_column_to_knowledge_base_categories_table.php:45]: There is no `after` in Postgres, so we can remove this here.

We are going to do 1 or 2 more Sprints where migration changes are allowed. So feel free to move this change into the first migration file for this table.

**Review by Orrison — CHANGES_REQUESTED** [review]

**Review by mxm1070 — COMMENTED** [review]
@Orrison are we ever going to use other icon packs? If so, it might be worth it to make this slightly more generic.

**Comment by Orrison** [comment]
> @Orrison are we ever going to use other icon packs? If so, it might be worth it to make this slightly more generic.

@mxm1070, not as far as I am aware of right now. Good thought though.

@danharrin, if you feel that it is easy/quick to make this more generic for future-proofing, then go ahead and take care of that now. If it is too much of a hassle. We can tackle that if/when we decide to use other icon packs.

**Comment by danharrin** [comment]
It's future proof as we store the name of the library in the icon name. ATM I am only using Heroicons in the code since there are other custom icons registered.

File: app-modules/knowledge-base/database/migrations/2024_02_12_123144_add_icon_column_to_knowledge_base_categories_table.php (hunk 1 above)
Comment (danharrin) [reply app-modules/knowledge-base/database/migrations/2024_02_12_123144_add_icon_column_to_knowledge_base_categories_table.php]: thanks, merged

**Review by Orrison — APPROVED** [review]

**Pull request merged**